src/components/NcBlurHash/NcBlurHash.vue

```diff
@@ -0,0 +1,260 @@
+
+```vue
+	<template>
+		<div class="wrapper">
+			<img :class="loaded ? 'shown-image' : 'hidden-visually'"
+				alt=""
+				src="favicon-touch.png">
+			<NcBlurHash v-if="!loaded"
+				class="shown-image"
+				:hash="blurHash" />
+			<NcButton @click="loaded = !loaded">Toggle blur hash</NcButton>
+		</div>
+	</template>
+	<script>
+	export default {
+		data() {
+			return {
+				loaded: false,
+				blurHash: 'M8CR]OkDD%kD9ZtRayofaykC00ay$_ay~T',
+			}
+		},
+	}
+	</script>
+	<style scoped>
+		.wrapper {
+			display: flex;
+			flex-direction: row;
+			align-items: center;
+			gap: 12px;
+		}
+		.shown-image {
+			width: 150px;
+			height: 150px;
+			border-radius: 24px;
+		}
+	</style>
+```
+</docs>
+
+<script setup>
+import { decode } from 'blurhash'
+import { ref, watch, nextTick } from 'vue'
+import { logger } from '../../utils/logger.ts'
+import { preloadImage } from '../../functions/preloadImage/index.ts'
+
+const props = defineProps({
+	/**
+	 * The blur hash value to use.
+	 */
+	hash: {
+		required: true,
+		type: String,
+	},
+
+	/**
+	 * This is normally not needed, but if this blur hash is not only intended
+	 * for decorative purpose, descriptive text should be passed for accessibility.
+	 */
+	alt: {
+		type: String,
+		default: '',
+	},
+
+	/**
+	 * Optional an image source to load, during the load the blur hash is shown.
+	 * As soon as it is loaded the image will be shown instead.
+	 */
+	src: {
+		type: String,
+		default: '',
+	},
+})
+
+const emit = defineEmits([
+	/**
+	 * Emitted when the image (`src`) has been loaded.
+	 */
+	'load',
+])
+
+const canvas = ref()
+const imageLoaded = ref(false)
+
+// Redraw when hash has changed
+watch(() => props.hash, drawBlurHash)
+// Redraw if image loaded again - also draw immediate on mount
+watch(imageLoaded, () => {
+	if (imageLoaded.value === false) {
+		// We need to wait one tick to make sure the canvas is in the DOM
+		nextTick(() => drawBlurHash())
+	}
+}, { immediate: true })
+
+// Preload image on source change
+watch(() => props.src, () => {
+	imageLoaded.value = false
+	if (props.src) {
+		preloadImage(props.src)
+			.then((success) => {
+				imageLoaded.value = success
+				emit('load', success)
+			})
+	}
+}, { immediate: true })
+
+/**
+ * Render the BlurHash within the canvas
+ */
+function drawBlurHash() {
+	if (imageLoaded.value) {
+		// skip
+		return
+	}
+
+	if (!props.hash) {
+		logger.error('Invalid BlurHash value')
+		return
+	}
+
+	if (canvas.value === undefined) {
+		// Should never happen but better safe than sorry
+		logger.error('BlurHash canvas not available')
+		return
+	}
+
+	const { height, width } = canvas.value
+	const pixels = decode(props.hash, width, height)
+
+	const ctx = canvas.value.getContext('2d')
+	if (ctx === null) {
+		logger.error('Cannot create context for BlurHash canvas')
+		return
+	}
+
+	const imageData = ctx.createImageData(width, height)
+	imageData.data.set(pixels)
+	ctx.clearRect(0, 0, width, height)
```

Comment:
Is it needed? I would have assumed that the cleared context will be overridden by the image data.

Reply:
Not if we use the canvas size

```suggestion
```